### PR TITLE
Nasser article about docker

### PR DESCRIPTION
**Pull request by abdounasser202**

### CODE REVIEW

File: content/post/articles/docker-basics-and-commands.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+In simple terms, as explained above, a Docker image is a template that contains the application and all the dependencies required to run that application on Docker.
+
+On the other hand, a Docker container is a logical entity. In more precise terms, it is a running instance of the Docker image.
+
+A container is a standard software entity that bundles code and all its dependencies so that the application runs quickly and reliably from one environment to another. A Docker container image is a lightweight, self-contained, executable package of software that includes everything needed to run an application: code, runtime, system tools, system libraries, and settings.
+
+## Dockerfile
+
+The Dockerfile is a text document that contains all the commands a user could call on the command line to assemble a docker image.
```
Comment (RMPR) [inline content/post/articles/docker-basics-and-commands.md:38]: s/document/file/

File: content/post/articles/docker-basics-and-commands.md [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+The Dockerfile is essentially the instructions for building the image. The advantage of a Dockerfile over simply storing the binary image (or a snapshot/template in other virtualisation systems) is that the automatic builds ensure that you have the latest version available.
+
+```bash
+# Example dockerfile to install Odoo
+# Note that this Odoo image is based on the debian:buster-slim image
+FROM odoo/odoo:14.0
+LABEL maintainer="You Name. <your@email.com>"
```
Comment (RMPR) [inline content/post/articles/docker-basics-and-commands.md:45]: Typo here, "Your" I guess?

File: content/post/articles/docker-basics-and-commands.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+RUN pip3 install --upgrade pip && pip3 install -r requirements_custom.txt
+
+WORKDIR /odoo/odoo
+
+USER odoo
+```
+
+## Docker-Compose
+
+Docker-Compose is a tool for defining and running multi-container Docker applications. With Docker-Compose, you use a YAML file to configure the services of your application before running it on docker.
```
Comment (RMPR) [inline content/post/articles/docker-basics-and-commands.md:58]: A bit of a nitpick here, but Docker-Compose shouldn't have an hyphen.

File: content/post/articles/docker-basics-and-commands.md [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+## Volumes
+
+In the previous **docker-compose.yml** file, you can see the volumes section. This is because Docker volumes are file systems mounted on Docker containers to preserve the data generated by the running container. Volumes are stored on the host, independent of the container's lifecycle. This allows users to backup data and easily share file systems between containers.
+
+## Networking
+
+Docker networking allows you to attach a container to as many networks as you like. For example, you can attach the Odoo server to the database server. When using the **docker-compose.yml** file, networking is done when you use the **depends_on** keyword in the file.
+
+If you want to do it manually for example by attaching your running web application to a **my_bridge**, you can type
```
Comment (RMPR) [inline content/post/articles/docker-basics-and-commands.md:124]: something missing here, maybe: a network named **my_bridge**?

**Review by RMPR — CHANGES_REQUESTED** [review]

File: content/post/articles/docker-basics-and-commands.md [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+
+This article aims to help you understand the basics of docker. First, I explain the concepts behind docker, then I explain how to install it and finally I give you some useful commands to use docker.
+
+# Containers VS Virtual Machines
+
+Virtual machines allow multiple operating systems to run on the hardware of a single physical server, while containers allow multiple applications running the same operating system to be deployed on a single virtual machine or server.
+
+# What is Docker
+
+Docker is an open source platform used to create, run and deploy applications in containers. Docker allows you to separate applications from your infrastructure so you can deliver software quickly. With Docker, you can manage your infrastructure in the same way as your applications.
```
[the remark this answers is not in the document]
Comment (abdounasser202) [reply content/post/articles/docker-basics-and-commands.md:16]: There are an enterprise version and open source one (licence apache 2.0)

File: content/post/articles/docker-basics-and-commands.md (hunk 1 above)
Comment (abdounasser202) [reply content/post/articles/docker-basics-and-commands.md:38]: what do you mean ?

File: content/post/articles/docker-basics-and-commands.md (hunk 2 above)
Comment (abdounasser202) [reply content/post/articles/docker-basics-and-commands.md:45]: thanks

File: content/post/articles/docker-basics-and-commands.md (hunk 3 above)
Comment (abdounasser202) [reply content/post/articles/docker-basics-and-commands.md:58]: thanks

File: content/post/articles/docker-basics-and-commands.md (hunk 4 above)
Comment (abdounasser202) [reply content/post/articles/docker-basics-and-commands.md:124]: thanks

File: content/post/articles/docker-basics-and-commands.md [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+date: 2021-09-28T00:16:28+01:00
+categories: [tutorial, docker, newbie]
+authors: [nab]
+---
+
+This article aims to help you understand the basics of docker. First, I explain the concepts behind docker, then I explain how to install it and finally I give you some useful commands to use docker.
+
+# Containers VS Virtual Machines
+
+Virtual machines allow multiple operating systems to run on the hardware of a single physical server, while containers allow multiple applications running the same operating system to be deployed on a single virtual machine or server.
```
[the remark this answers is not in the document]
Comment (abdounasser202) [reply content/post/articles/docker-basics-and-commands.md:12]: updated

File: content/post/articles/docker-basics-and-commands.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (abdounasser202) [reply content/post/articles/docker-basics-and-commands.md:16]: added one more phrase

File: .gitignore [hunk 7]
```diff
@@ -17,3 +17,6 @@
 #hugo
 public
 _gen
+
+# IDE
+.idea/
```
Comment (elhmn) [inline .gitignore:22]: ❤️

**Review by elhmn — APPROVED** [review]
What a great Article, and docker thank you for setting up docker 🙇‍♂️  ❤️

File: content/post/articles/docker-basics-and-commands.md (hunk 1 above)
Comment (RMPR) [reply content/post/articles/docker-basics-and-commands.md:38]: I meant you should replace "text document" by "text file" because it's more correct imo

**Pull request merged**